src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2171,6 +2241,57 @@ static void uct_ib_mlx5dv_check_dm_ksm_reg(uct_ib_mlx5_md_t *md)
 #endif
 }
 
+static void uct_ib_mlx5dv_check_direct_nic(struct ibv_context *ctx,
+                                           uct_ib_device_t *dev,
+                                           uct_ib_mlx5_md_t *md,
+                                           const uct_ib_md_config_t *md_config)
+{
+#ifdef HAVE_DIRECT_NIC
+    char sys_path[PATH_MAX];
+    char dev_name[64];
+    int ret;
+    ucs_status_t status;
+
+    if (!md_config->ext.direct_nic) {
+        ucs_debug("%s: Direct NIC support disabled",
+                  uct_ib_device_name(&md->super.dev));
+        return;
+    }
+
+    ret = mlx5dv_get_data_direct_sysfs_path(ctx, sys_path, sizeof(sys_path));
+    if (ret == 0) {
+        /* Create a DMA specific device from topology perspective */
+        snprintf(dev_name, sizeof(dev_name), "%s_dma",
+                 uct_ib_device_name(&md->super.dev));
+        md->direct_nic_sys_dev = ucs_topo_get_sysfs_dev(dev_name, sys_path, 0);
+
+        status = ucs_topo_sys_device_set_sys_dev_aux(dev->sys_dev,
+                                                     md->direct_nic_sys_dev);
+        ucs_assertv_always(status == UCS_OK, "set sys_dev_aux failed: %s",
+                           ucs_status_string(status));
+        md->super.dev.flags |= UCT_IB_DEVICE_FLAG_DIRECT_NIC;
+    } else {
+        sys_path[0]            = '\0';
+        md->direct_nic_sys_dev = UCS_SYS_DEVICE_ID_UNKNOWN;
+        (void)ucs_topo_sys_device_set_sys_dev_aux(dev->sys_dev,
+                                                  md->direct_nic_sys_dev);
+    }
+
+    ucs_debug("%s: Direct NIC %s supported ret=%d sys_path='%s%s' "
```

Comment:
smth like 
ret = mlx5dv_get_data_direct_sysfs_path ...
if (ret != 0) {
ucs_debug("%s: mlx5dv_get_data_direct_sysfs_path(%s failed: %m", ib_dev_name, sys_path...)
return
}

ucs_debug("%s: direct nic is supported, sys_dev %s", ib_dev_name, sys_dev_name")